docs/csharp/csharp-7.md

```diff
@@ -16,64 +16,531 @@ ms.assetid: fd41596d-d0c2-4816-b94d-c4d00a5d0243
-the C# compiler team. The language is truly open.
-
-The remainder of this topic provides an overview 
+The remainder of this topic discusses each of the features. For each feature,
+you'll learn the reasoning behind it. You'll learn the syntax. You'll see
+some sample scenarios where using the new feature will make you more 
+productive as a developer. 
 
 ## `out` variables
 
+The existing syntax that supports `out` parameters has been improved
+in this version.  
+
+Previously, you would need to separate the declaration of the out variable
+and its initialization into two different statements:
+
+[!code-csharp[OutVariableOldStyle](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#03_OutVariableOldStyle "classic out variable declaration")]
+
+You can now declare `out` variables in the argument list of a method call,
+rather than writing a separate declaration statement:
+
+[!code-csharp[OutVariableDeclarations](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#01_OutVariableDeclarations "Out variable declarations")]
+
+You may want to specify the type of the `out` variable for clarity,
+as shown above. However, the language does support using an implicitly
+typed local variable:
+
+[!code-csharp[OutVarVariableDeclarations](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#02_OutVarVariableDeclarations "Implicitly typed Out variable")]
+
+<!--
+Add a sample at RC that shows how if statements
+scope out variables.
+-->
+
+This small language change improves your productivity in several ways:
+* The code is easier to read. 
+    - You declare the out variable where you use it, not on another line above.
+* No need to assign an initial value.
+    - By declaring the `out` variable where it is used in a method call, you can't accidentally use it before it is assigned.
+* The scope of the variable is minimized.
+    - The `out` variable is scoped to the enclosing block. By minimizing the scope, you avoid name collisions with other variables and you minimize the chance of misuse.
+
+The most common use for this feature will be the `TryParse` pattern. In this
+pattern, a method returns a `bool` indicating success or failure and an
+`out` variable that provides the result if the method succeeds.
 
 ## Tuples
 
-### Named and unnamed Tuples
+C# provides a rich used for classes and structs provides to explain
+your design intent. But sometimes that rich syntax requires extra
+work with minimal benefit. You may often write methods that need a simple
+structure containing more than one data element. To support these scenarios
+*tuples* were added to C#. Tuples are lightweight data structures
+that contain multiple elements.  Tuples are lightweight data structures
+that contain multiple fields to represent the data members of the tuple.
+The fields are not validated, and you cannot define your own methods
+
+> [!NOTE]
+> Tuples were available before C# 7, but had many limitations. Most importantly,
```

Comment:
May want to clarify they were available as an API, not as a feature of the language.